packages/application-components/src/components/modal-pages/form-modal-page/form-modal-page.tsx

```diff
@@ -26,27 +37,47 @@ type CommonProps = {
   dataAttributesPrimaryButton?: { [key: string]: string };
   dataAttributesSecondaryButton?: { [key: string]: string };
 };
-type PropsWithCustomControls = CommonProps & {
-  customControls: React.ReactNode;
-};
-type PropsWithoutCustomControls = CommonProps & {
-  labelSecondaryButton: Label;
-  labelPrimaryButton: Label;
-  onSecondaryButtonClick: (event: React.SyntheticEvent) => void;
-  onPrimaryButtonClick: (event: React.SyntheticEvent) => void;
-};
-type Props = PropsWithCustomControls | PropsWithoutCustomControls;
+type Props = CommonProps & WithCustomControls & WithoutCustomControls;
+type PropsWithCustomControls = CommonProps & Required<WithCustomControls>;
+type PropsWithoutCustomControls = CommonProps & Required<WithoutCustomControls>;
 const defaultProps: Pick<
-  PropsWithoutCustomControls,
+  Props,
   'labelPrimaryButton' | 'labelSecondaryButton'
 > = {
   labelPrimaryButton: buttonMessages.confirm,
   labelSecondaryButton: buttonMessages.cancel,
 };
 
 // Type-guard validation for the correct props, based on the existence `customControls`
-const hasCustomControls = (props: Props): props is PropsWithCustomControls =>
+const hasCustomControls = (
+  props: PropsWithCustomControls | PropsWithoutCustomControls
+): props is PropsWithCustomControls =>
   'customControls' in props && props.customControls !== undefined;
+const getConditionalProps = (props: Props) => {
+  if ('customControls' in props && props.customControls !== undefined) {
```

Comment:
Can't we use the `hasCustomControls` fn here, since it's the same condition?

Reply:
No because the type of the `props` argument is different